create_statement_creator/athena_schema.py

```diff
@@ -36,16 +37,39 @@ def create_athena_column_schema(avro_schema) -> str:
         return f'struct<{field_schema_concatenated}>'
 
     elif type(avro_schema) == UnionSchema:
-        # pick the first schema which is not null
         union_schemas_not_null = [s for s in avro_schema.schemas if s.type != 'null']
-        if len(union_schemas_not_null) > 0:
+        contains_null_schema = len([s for s in avro_schema.schemas if s.type == 'null']) > 0
+        if contains_null_schema and len(union_schemas_not_null) == 1:
             return create_athena_column_schema(union_schemas_not_null[0])
-        else:
+        elif contains_null_schema and len(union_schemas_not_null) == 0:
             raise Exception('union schemas contains only null schema')
 
+        union_schemas = []
+        i = 0
+        for schema in union_schemas_not_null:
+            union_schemas.append(f'member{i}:{create_athena_column_schema(schema)}')
+            i += 1
+        members_concatenated = ','.join(union_schemas)
+        return f'struct<{members_concatenated}>'
```

Comment:
Can you provide a link to a documentation or an example of Glue behaving like this in case of a union of multiple types? I am not able to verify this behavior with a Google search, and that it works in Athena.

Also, I find it difficult to understand what the code does. Is the following correct? 

- If the union contains null and exactly one non-null, return the non-null
- If it contains more than one non-null, return a struct with member0, member1 etc of the non-nulls
- If it contains only null, raise an error

Maybe one can refactor the code a little?

Reply:
I found out this Glue behavior when I ran a crawler on a data set that has a union with a single non-null data type defined.

Since Athena does not support tables with Hive UNIONTYPE columns, this struct "trick" was a way to make the data queryable.

I'm unsure how it scales to multiple data types in the union, though.

I'll see if I can add such a case to the example schema or separately and provide some example data.